Sources/ComposableArchitecture/ViewStore.swift

```diff
@@ -83,6 +83,16 @@ public final class ViewStore<State, Action>: ObservableObject {
       }
   }
 
+  fileprivate init(
+    _send: @escaping (Action) -> Void,
+    _state: CurrentValueRelay<State>,
+    viewCancellable: AnyCancellable? = nil
```

Comment:
`viewCancellable` is only really relevant to the root `ViewStore` so I'm not sure it needs to be included. I originally included `objectWillChange` in case the copied view store was observed in some way. This actually makes me wonder if creating a new `ViewStore` may break `.binding()` animations.

Reply:
Dropping `viewCancellable` was indeed something I considered. I quickly tried to find cases where it would be needed and I found none yet, but I opted for making a _bona fide_ `ViewStore` in any case. As I'm getting it, keeping `viewCancellable` protects against the user sneaking it outside `SwiftUI` in `onAppear` or `task` for example (I may have done things like that in the past!). Without the cancellable, the escaped instance's state will unsync. Do you see issues in duplicating the `viewCancellable`?

I'll check with animations later, but they're still working in the "Animation" case study where an animated binding is used to toggle "Big mode".

Reply:
Ah, the `CurrentValueRelay` is a class, so it will be updated by the parent. I'll need to think more about it then!

Reply:
You're probably right. The `AnyCancellable` is only a reference to the existing subscription from the parent, and the `CurrentValueRelay` is the same instance, so there is no need to keep one additional reference to the subscription. I'll update the PR in this sense if it doesn't break anything.

Reply:
> the CurrentValueRelay is a class

Yeah. The `viewCancellable` will only cancel the sink subscription to the `Store` which is linked to the original `ViewStore`. Hypothetically this could be contained in a single class which would then allow `ViewStore` to be independently copied by passing that reference around. Essentially making `ViewStore` more like a wrapper. I just didn't want to introduce too much of a change.

> they're still working in the "Animation" case study where an animated binding is used to toggle "Big mode"

I confirmed animations were working too.

Reply:
I've updated and reverted the change! As long as the `@ObservedObject` `ViewStore` is alive, there is no issue, but if it goes away and someone still holds the `viewStore` somewhere outside of `SwiftUI`, updating will stop.
I think it should be kept for this reason. Don't you think?

Reply:
> updating will stop.

It will stop regardless. If the `@ObservedObject` `ViewStore` doesn't exist. The subscription becomes a no-op:

```swift
self.viewCancellable = store.state
  .removeDuplicates(by: isDuplicate)
  .sink { [weak self] in
    guard let self = self else { return }
    self.objectWillChange.send()
    self._state.value = $0
  }
```

This can be fixed by capturing `ObservableObjectPublisher` and `CurrentValueRelay` separately (or wrapping them up together). But like I said, I didn't suggested it to avoid introducing additional changes.

Reply:
essentially something like this: 

```swift
  public init(
    _ store: Store<State, Action>,
    removeDuplicates isDuplicate: @escaping (State, State) -> Bool
  ) {
    self._send = { store.send($0) }
    self._state = CurrentValueRelay(store.state.value)

    self.viewCancellable = store.state
      .removeDuplicates(by: isDuplicate)
      .sink { [weak objectWillChange = self.objectWillChange, weak _state = self._state] in
        guard let objectWillChange = objectWillChange, let _state = _state else { return }
        objectWillChange.send()
        _state.value = $0
      }
  }

  private init(viewStore: ViewStore<State, Action>) {
    self._send = viewStore._send
    self._state = viewStore._state
    self.objectWillChange = viewStore.objectWillChange
    self.viewCancellable = viewStore.viewCancellable
  }
```

That would then work the way you're describing, and holding on to `viewCancellable` would be necessary.

Reply:
> It will stop regardless.

You're right, I missed that.

Your capturing implementation seems fine to me, and it doesn't look much more complex than the current one. You're just capturing more narrowly. The `objectWillChange` translocation shouldn't be problematic as this publisher doesn't reference `self` and we don't let the framework synthetize-it to begin with.
I'll refactor in this spirit!
You should get all the credit for this PR!

Reply:
Cool. The updated init looks good too. 

> You're just capturing more narrowly

Which I thought could introduce additional overhead, but it's likely nothing. 

> You should get all the credit for this PR!

Team effort :)

Reply:
I've updated the PR. One novel situation is having two different `@ObservedObject` observing different instances, but sharing the same `objectWillChange` publisher. It can happen if the `viewStore` is passed as an argument to an inner view that installs it as `@ObservedObject`. It should most likely work, but I'll check if SwiftUI hasn't specific optimizations around `@ObservedObject`.

Reply:
> It can happen if the viewStore is passed as an argument to an inner view that installs it as @ObservedObject

That's the main reason for sharing the objectWillChange publisher. Even though it's probably not applicable because if the parent view changes the child view will be passed a new view store to use with `@ObservedObject`.

Reply:
> Which I thought could introduce additional overhead, but it's likely nothing.

Benchmarking shows no difference, in any case.